File: 1370. Increasing Decreasing String/Solution.py
import collections
import string


class Solution:
    def sortString(self, s: str) -> str:
        counter, res = collections.Counter(s), []
        while counter:
            for traverse in string.ascii_lowercase, reversed(string.ascii_lowercase):
                res += [c for c in traverse if c in counter]
                counter -= dict.fromkeys(counter, 1)
        return ''.join(res)

File: 1370. Increasing Decreasing String/Solution.java
class Solution {
    public String sortString(String s) {
        int[] map = new int[26];
        for (char c : s.toCharArray())
            map[c - 'a']++;

        StringBuilder res = new StringBuilder();
        while (res.length() != s.length()) {
            for (int i = 0; i < 26; i++)
                if (map[i]-- > 0)
                    res.append((char)('a' + i));
            for (int i = 25; i >= 0; i--)
                if (map[i]-- > 0)
                    res.append((char)('a' + i));
        }
        return res.toString();
    }
}
